src/lib/pickles/test/test_wrap.ml

```diff
@@ -0,0 +1,334 @@
+      (module Impls.Step)
+      ~feature_flags:full_features ~challenge:Challenge.typ
+      ~scalar_challenge:Challenge.typ
+      ~dummy_scalar_challenge:
+        (Kimchi_backend_common.Scalar_challenge.create
+           Limb_vector.Challenge.Constant.zero )
+      (Shifted_value.Type1.typ Field.typ)
+      (Import.Branch_data.typ
+         (module Impl)
+         ~assert_16_bits:(Step_verifier.assert_n_bits ~n:16) )
+  in
+
+  (* Use deferred_values_typ and the constant helper to prepare deferred_values
+     for use in the circuit.  We change some [Opt.t] to [Option.t] because that is
+     what Type.t is configured to accept. *)
+  let deferred_values =
+    constant deferred_values_typ
+      { deferred_values with
+        plonk =
+          { deferred_values.plonk with
+            joint_combiner =
+              Opt.to_option_unsafe deferred_values.plonk.joint_combiner
+          }
+      }
+  (* Prepare all of the evaluations (i.e. all of the columns in the proof that we open)
+     for use in the circuit *)
+  and evals =
+    constant
+      (Plonk_types.All_evals.typ ~num_chunks:1
+         (module Impls.Step)
+         full_features )
+      { evals =
+          { public_input = x_hat_evals; evals = proof.proof.openings.evals }
+      ; ft_eval1 = proof.proof.openings.ft_eval1
+      }
+  in
+
+  (* Run the circuit without generating a proof using run_and_check *)
+  Impls.Step.run_and_check (fun () ->
+      (* Set up the step sponge from the wrap sponge -- we cannot use the same poseidon
+         sponge in both step and wrap because they have different fields.
+
+         In order to continue the Fiat-Shamir heuristic across field boundaries we use
+         the wrap sponge for everything in the wrap proof, squeeze it one final time and
+         expose the squoze value in the public input to the step proof, which absorbs
+         said squoze value into the step sponge. :-) This means the step sponge has absorbed
+         everything from the proof so far by proxy and that is also over the native field! *)
+      let res, _chals =
+        let sponge =
+          let open Step_main_inputs in
+          let sponge = Sponge.create sponge_params in
+          Sponge.absorb sponge
+            (`Field (Impl.Field.constant sponge_digest_before_evaluations)) ;
+          sponge
+        in
+
+        (* Call finalisation with all of the required details *)
+        Step_verifier.finalize_other_proof
+          (module Nat.N0)
+          ~step_domains:
+            (`Known [ { h = Pow_2_roots_of_unity vk.domain.log_size_of_group } ])
+          ~zk_rows:3 ~sponge ~prev_challenges:[] deferred_values evals
+      in
+
+      (* Read the boolean result from the circuit and make it available
+         to the OCaml world. *)
+      Impls.Step.(As_prover.(fun () -> read Boolean.typ res)) )
+  |> Or_error.ok_exn
+
+(* Common srs value for all tests *)
+let srs =
+  Kimchi_bindings.Protocol.SRS.Fp.create (1 lsl Common.Max_degree.step_log2)
```

Comment:
Question: isn't there a cached SRS for tests in `mina` similar to `proof-systems/srs/`?

Reply:
I don't think there is.